packages/completer/src/tokens.ts

```diff
@@ -206,5 +206,5 @@ export interface IProviderReconciliator {
   shouldShowContinuousHint(
     completerIsVisible: boolean,
     changed: SourceChange
-  ): boolean;
+  ): Promise<boolean>;
```

Comment:
I think that with little effort we could have a union of a promise or boolean for better backward compatibility. I find it unlikely that downstreams modify reconciliator since it was just recently released and is not the documented as an extension point, but if we are to improve extension ecosystem experience it probably starts with such little decisions.

Reply:
Could you give me a hint about the implementation?

Reply:
I was thinking that it would be just changing the return type to union of promise and a boolean, without changed to implementation, but if that's not that simple then maybe it's not worth it after all...

Reply:
Yep, I think it's not that simple, and returning `boolean | Promise<boolean>` would confuse the downstream usage.

Reply:
To be fair I think this would not be used downstream much, but then this argument is also why I don't think we need to change it :) Marking as resolved, thank you for engaging in brainstorming this one!